Visualization/generate_BackBLAST_heatmap.R

```diff
@@ -0,0 +1,618 @@
+#!/usr/bin/env Rscript
+# generate_BackBLAST_heatmap.R
+# Copyright Lee Bergstrand and Jackson M. Tsuji, 2019
+# Plots a newick treefile and BLAST table together as a phylogenetic tree and heatmap
+
+# Load libraries
```

Comment:
@LeeBergstrand `warn.conflicts = FALSE` just masks warnings where function names from two different packages overlap. This is not an issue because I specify the namespace of all non-base functions. The namepace warnings are very long and are annoying to see every time the script runs. Do you agree that it is okay to hide them? Alternatively, I just did some reading and saw that I can globally disable those warnings by specifying `options(conflicts.policy(warn = FALSE))` instead of including the `warn.conflicts` flag with each package load.

Thanks for pointing out `quietly = TRUE` -- that was not necessary, and I've deleted it.

`suppressMessages(library(ggtree))` blocks a long package load message:
```
ggtree v1.14.2  For help: https://guangchuangyu.github.io/software/ggtree

If you use ggtree in published research, please cite the most appropriate paper(s):

◉ Guangchuang Yu, David Smith, Huachen Zhu, Yi Guan, Tommy Tsan-Yuk Lam. ggtree: an R package for visualization and annotation of phylogenetic trees with their covariates and other associated data. Methods in Ecology and Evolution 2017, 8(1):28-36, doi:10.1111/2041-210X.12628

◉ Guangchuang Yu, Tommy Tsan-Yuk Lam, Huachen Zhu, Yi Guan. Two methods for mapping and visualizing associated data on phylogeny using ggtree. Molecular Biology and Evolution 2018, accepted. doi: 10.1093/molbev/msy194
```
I did some reading and found I can use the more specific suppressor `suppressPackageStartupMessages()`. I've updated to this function now.

I do agree that we should get rid of `suppressMessages(library(ggplot2))`. That blocks a message about rlang:
```
Registered S3 methods overwritten by 'ggplot2':
  method         from 
  [.quosures     rlang
  c.quosures     rlang
  print.quosures rlang
```
This looks like this is a legit error that the ggplot team has worked on.

Thoughts?

Reply:
Also, should I be specifying the namespace of base functions as well, e.g, `base::read.table()`?

Or there might be a way to prevent any loaded packages from masking the namespace of base packages.

Reply:
@jmtsuji Yes do ```options(conflicts.policy(warn = FALSE))``` and put a block comment on top of it explaining why. Also, does CLI has a ```--debug``` flag that sets the logging to debug mode? You could change the ```warn = FALSE``` to TRUE if in debug mode.

Reply:
> I did some reading and found I can use the more specific suppressor suppressPackageStartupMessages(). I've updated to this function now.

Sounds good.

Reply:
> I do agree that we should get rid of suppressMessages(library(ggplot2))
@jmtsuji Yes do this.

Reply:
> Also, should I be specifying the namespace of base functions as well, e.g, `base::read.table()`?
> 
> Or there might be a way to prevent any loaded packages from masking the namespace of base packages.

@jmtsuji If the function is a built-in of the R standard library (I assume base are the functions built into R), then I would assume that leaving ```base::``` off would be appropriate. You would probably only want to add a namespace if its non-default.

Reply:
> @jmtsuji Yes do `options(conflicts.policy(warn = FALSE))` and put a block comment on top of it explaining why. Also, does CLI has a `--debug` flag that sets the logging to debug mode? You could change the `warn = FALSE` to TRUE if in debug mode.

Just looked into this. Unfortunately, this is a brand new feature (R 3.6.0) and is not available in R 3.5.1, which I am currently working with to satisfy all conda dependencies. We'll have to wait a couple months before taking advantage of `options(conflicts.policy(list(warn = FALSE)))`. For now, I've left a note explaining why the `warn.conflicts = FALSE` flags are there. No easy way to change this during debug right now, unfortunately.

Reply:
> @jmtsuji If the function is a built-in of the R standard library (I assume base are the functions built into R), then I would assume that leaving `base::` off would be appropriate. You would probably only want to add a namespace if its non-default.

@LeeBergstrand Note that some R packages override the standard library function names. E.g., `base::filter()` and `dplyr::filter()` do very different things. (Is this typical for programming languages?) Thus, you run a risk that your base package might have a namespace conflict.

Reply:
P.S. In my case, I've already checked all namespace issues (via `warn.conflicts = TRUE`) and confirmed that I do not have any namespace conflicts with base R functions I call. I'm just curious about best practices.

Reply:
@LeeBergstrand Okay to merge this pull request now?